### DISCUSSION TITLE
Adapt rejection of workflowitems

### DISCUSSION THREAD

**Discussion by Stezido** [openkfw/TruBudget#1300] [Ideas]
### Description
Rejecting an workflowitem closes the workflowitem and the workflow continues.
For better user experience a rejection should not be a continuation of the workflow, instead the workflowitem must be edited before.

**Comment by Stezido** [comment]
Should the workflow be stopped in case of a rejection?
Yes - Good Usability. Updates can be done after rejection. Breaking change.
No - Each step should be an own workflowitem including rejections to keep all action in the history. Usability is worse.

Should a rejection be part of the history/ stored on blockchain?
Yes - Traceability - every event should be part of the history.
No - ?

How it currently works from api side?
Rejection = Event workflowitem.close with reject Reason property
Approve/Close = Event workflowitem.close without reject Reason property

**Reply by LukasKM** [reply]
Add 1) Should the workflow be stopped in case of a rejection?
Can you please specify what “updates can be done after rejection”… mean?
For the Tunisia setup, NO is preferable, because it gives the flexibility to deal with the overall workflow (sub-project) depending on the nature of the process, and avoids any breaking changes.

Add 2) Should a rejection be part of the history/ stored on blockchain?
Yes - for traceability. Each critical activity/event should be documented on the BC.

**Reply by RMutius** [reply]
In general, workflows shall be stopped in case of a rejection. Workflows represent a logical, step-by-step succession of workflow items, with each workflow item building on the before item, i.e. requiring it to be marked as completed. Therefore, continuing a workflow even though a prior step was rejected would not be logical. E.g. if you have a requirement for a payment in your flow, you would not proceed with the payment even though the requirement was not fulfilled, i.e. rejected.

In this scenario, one should have the possibility to re-send an approval request for a workflow item once it was denied. E.g. if you would submit documents for approval, you could - after denial - send an updated version, i.e. request the new approval of the same workflow item. This would allow to set up the workflows in the project beginning. Otherwise you would have to constantly, modify the workflow items and add steps in case of rejection of a workflowitem. 

These rejection history should be part of the history to have a full and transparent documentation.
